libs/libc/stdio/lib_vprintf.c

```diff
@@ -49,7 +52,13 @@
 
 int vprintf(FAR const IPTR char *fmt, va_list ap)
 {
+#if CONFIG_NFILE_STREAMS > 0
   /* vfprintf into stdout */
 
   return vfprintf(stdout, fmt, ap);
+#else
+  /* vsyslog into debug channel */
+
+  return nx_vsyslog(LOG_INFO, fmt, &ap);
```

Comment:
It would be much better to use vdprintf()

Reply:
vdprintf() will behave exactly like printf, but will not depend on the FILE stdout.  That is a better choice.

Reply:
syslog() has other properties and may not output of fd == 0 as is REQUIRED for printf().

Reply:
If you want that kind of nonstandard behavior from printf for some reason. You probably could create a configuration option to redirect print to syslog.  But I think it is unacceptable as default behavior for printf() because it breaks it required behavior.